src/darwin/Framework/CHIPTests/MTRPerControllerStorageTests.m

```diff
@@ -2298,4 +2298,496 @@ - (void)testSubscriptionPool
+                    XCTAssertNotNil(persistedAttributes);
+                    for (NSNumber * attribute in testAttributes) {
+                        XCTAssertTrue([persistedAttributes containsObject:attribute]);
+                    }
+                }
+            }
+        });
+
+        [subscriptionExpectation fulfill];
+    };
+
+    [device setDelegate:delegate queue:queue];
+
+    [self waitForExpectations:@[ subscriptionExpectation ] timeout:60];
+
+    dataVersionForIdentify = [NSNumber numberWithUnsignedLongLong:(dataVersionForIdentify.unsignedLongLongValue + 1)];
+
+    [testClusterDataValue removeObjectAtIndex:1];
+
+    NSArray<NSDictionary<NSString *, id> *> * attributeReport = @[ @{
+        MTRAttributePathKey : [MTRAttributePath attributePathWithEndpointID:testEndpoint clusterID:@(MTRClusterIDTypeIdentifyID) attributeID:@(MTRAttributeIDTypeGlobalAttributeAttributeListID)],
+        MTRDataKey : @ {
+            MTRDataVersionKey : dataVersionForIdentify,
+            MTRTypeKey : MTRArrayValueType,
+            MTRValueKey : testClusterDataValue,
+        }
+    } ];
+
+    XCTestExpectation * attributeDataReceivedExpectation = [self expectationWithDescription:@"Injected Attribute data received"];
+    XCTestExpectation * reportEndExpectation = [self expectationWithDescription:@"Injected Attribute data report ended"];
+    delegate.onAttributeDataReceived = ^(NSArray<NSDictionary<NSString *, id> *> * attributeReport) {
+        attributeReportsReceived += attributeReport.count;
+        XCTAssert(attributeReportsReceived > 0);
+        [attributeDataReceivedExpectation fulfill];
+    };
+
+    delegate.onReportEnd = ^{
+        // Make sure that the cluster data in the data storage is populated with cluster data for MTRClusterIDTypeIdentifyID cluster
+        // and has all attributes except attribute 1 which was deleted.
+        // We will page in the cluster data from storage to check the above.
+        dispatch_sync(self->_storageQueue, ^{
+            initialClusterIndex = [[controller.controllerDataStore _fetchClusterIndexForNodeID:deviceID endpointID:testEndpoint] mutableCopy];
+            XCTAssertNotNil(initialClusterIndex);
+
+            for (NSNumber * cluster in initialClusterIndex) {
+                MTRDeviceClusterData * clusterData = [controller.controllerDataStore _fetchClusterDataForNodeID:deviceID endpointID:testEndpoint clusterID:cluster];
+                XCTAssertNotNil(clusterData);
+                MTRClusterPath * path = [MTRClusterPath clusterPathWithEndpointID:testEndpoint clusterID:cluster];
+
+                if ([cluster isEqualToNumber:@(MTRClusterIDTypeIdentifyID)]) {
+                    MTRDeviceClusterData * data = [device _getClusterDataForPath:path];
+                    XCTAssertNotNil(data);
+                    XCTAssertNotNil(data.attributes);
+
+                    NSDictionary * dict = [data.attributes objectForKey:@(MTRAttributeIDTypeGlobalAttributeAttributeListID)];
```

Comment:
"dict" is a terrible name.  How about "attributeListValue"?

Reply:
done.